### ISSUE TITLE
Laser as a secondary Head option

### ISSUE THREAD

**Issue by flr09** [supermerill/SuperSlicer#2313]
This is a feature request for a Feature, Plugin or Macro in Super Slicer that modifies mostly existing options.
With this Feature Superslicer should generate Printable G-code, which is a partial repetition of generated Slices.

The request is based on an Idea of a Paper i found some time ago about priming deposited Layers with Laserlight for optimal Layer adhesion of FDM printed ABS parts.

for Printer:
- specyfiing the offset of the Laser on x/y
- specyfiing the focal distance to deposited Material/ buildplate 

for Print:
- specyfiing the intensity of the Laser on inner and outer Perimeter and Infill 
- specyfiing the speed on Perimeter on inner and outer Perimeter and Infill 
- specyfiing traveled distance or ON time of laser until next material deposit to reduce overheating of the Model
- specyfiing the length of points to strokes and distance to each other
- specyfiing the hight above Platform, when first layers are lased

actually not important: 
- specyfiing other pattern, like waves, zigzag, octagon
- overlapping
- 
- galvo support
- several diodes support

How it should Work:
The Paper purposed a red laser with about 500mW 
The Laser is hooked up to a Fan port on the motherboard and definded in the Printer config
The Laser is fixed on the x Gantry and has a fixed distance to the Toolhead.
The focus point of the laser should be similar or even to the extruder hight
The Laser performs a path on every or every other Layer to melt/ soften the last deposited layer.
The Distance of the Lased Path should not exeed a defined amount, as the Material could cool down too much before the next Layer is deposited.
The Laser could perform a dotted line or strokes in the middle of previous deposited Material for faster performace and to reduce upheating of material too much
The Laser could perform a wavy pattern to reach into the corners of actual deposited Material for optimal layer bonding of inner and outer perimeter  for optimal outer Layer adhesion.

What there is to find out:
how much powerper second is needed to soften the upper surface of semi cooled ABS Material
is there any shrinkage or tension in the Material
is there any warping
is the material annealed? 
is it possible to use fans in ABS?
is there any need for headed chamber while using laser?
is it possible to print ABS material cooler and with more pressure 

The printer, that uses a Laser fixture should have:
- Closed chamber with mechanical failsafe 
- Window with red block (Green or Blue) Filter 
- Ventilation 

as this is my first request i hope you understood what I´m requesting. 
Maybe you´re on something similar or have an Idea how to hook up the Diode laser.

best,

chris

**Comment by tylerinpdx** [comment]
Tldr for anyone, prints layer with abs and goes over it a second time with a laser to melt it better. That is so freaking cool but also extremely complicated, maybe you can even use the laser to melt stringing
